qiime2/sdk/plugin_manager.py

```diff
@@ -201,7 +201,7 @@ def _integrate_plugin(self, plugin):
                 )
 
             self.formats[name] = record
-        self.type_formats.extend(plugin.type_formats)
+        self.artifact_classes.extend(plugin.artifact_classes)
```

Comment:
This was missing originally, but we should do a similar check as the other properties to make sure duplicate registrations (across plugins) are forbidden.

Reply:
I added this test and validated that it works manually and reports a useful error message (`ValueError: Duplicate semantic type ('PCoAResults') defined in plugins: 'sapienns' and 'types'`). I didn't see where/if this type of functionality was being unit tested, so I didn't add unit tests for it now.

Reply:
There's no pleasant way to test for this, so eyeballing it is fine with me.